terraform/pipeline/workspace.py

```diff
@@ -5,18 +5,20 @@
 import sys
 
 REASON = os.environ["REASON"]
+
 # the name of the variable that Azure Pipelines uses for the source branch
 # depends on the type of run, so need to check both
 SOURCE = os.environ.get("OTHER_SOURCE") or os.environ["INDIVIDUAL_SOURCE"]
+
 TARGET = os.environ["TARGET"]
 
 # the branches that correspond to environments
-ENV_BRANCHES = ["dev", "test", "prod"]
+ENV_BRANCHES = {"main": "dev", "test": "test", "prod": "prod"}
```

Comment:
This is great to use a dictionary here instead of a list, couple of comments:

* The `key` of the items is the branch name; the `value` of the items is the workspace/environment name
* This allows us to have a different workspace name from the branch name
* It also means in the code, there's no need to have both an `environment` and `workspace` variable, since those were both used to help make the distinction between branch name and workspace name, that is now encoded in the dict

So a couple changes are still needed:

* Rename `environment` to `workspace`
* In both line 21 and line 24, use the dict to get the value for the `TARGET` or `SOURCE` branch
* Delete line 30, since `workspace` is now set appropriately
* Update line 32 `service_connection` to check `workspace` instead of `environment`
  * Further idea, create a new dict mapping workspace name to service connection name and use a lookup

I think this will really help simplify the code here.

Reply:
Thanks for the comments! I was still not completely clear about how the workspace/environment was being used but this explanation is really helpful, I think I got it now. Only `workspace` and `service_connection` are actually passed back to `azure-pipelines.yml` in `env_select` so these changes do make the code simpler 👍

Reply:
The `prod` branch should map to the `default` workspace

Reply:
Here's a reminder of the mapping: https://docs.calitp.org/benefits/deployment/infrastructure/#environments

Reply:
Maybe we rename `ENV_BRANCHES` to `WORKSPACES`?

Reply:
Thanks for catching that the `prod` branch should map to the `default` workspace!
I renamed `ENV_BRANCHES` to `WORKSPACES` to match the naming pattern of `service_connections` in line 31 but we can change the naming if we find something more descriptive.